### PR TITLE
Adding prometheus operator

### PR DESCRIPTION
**Pull request by nakfour**
This PR is to migrate the prometheus component. This does require the operatorgroup.yaml that was added to the Grafana PR, please make sure you manually install it. To test this
1. Install this component
kfctl build --file=kfdef/kfctl_openshift.yaml -V
kfctl apply --file=kfdef/kfctl_openshift.yaml -V
2. Go to prometheus portal from the routes and check Status-->Targets
3. Install opendatahub from the  community operator in namespace opendatahub.
4. Edit the Spark Service "spark-operator-metrics" to include name in the port
```spec:
  ports:
    - name: web
      protocol: TCP
      port: 8080
      targetPort: 8080
```
5. go back to Prometheus Status--->Targets, and re-fresh. You should see the new target for Spark. You can also search for one of the Spark variables "log4j"
6. For components that expose a prometheus endpoint, they need to name the port "web" or create a `servicemonitor` with specifics to their endpoint with a label `team: opendatahub` since this is what our Prometheus instance is looking for.

### CODE REVIEW

File: prometheus/operator/base/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: ServiceMonitor
+metadata:
+  name: odhservicemonitor
+  labels:
+    team: opendatahub
+  namespace: opendatahub
+spec:
+  selector: {}
+  endpoints:
+    - port: web
```
Comment (crobby) [inline prometheus/operator/base/servicemonitor.yaml]: is "web" a fairly standard prometheus port name?  It seems to me like a port named "web" would be more inclined to be a web app?

Comment (nakfour) [reply prometheus/operator/base/servicemonitor.yaml]: @crobby thats what is in the example since a lot use 8080 and yes it is an http endpoint

Comment (crobby) [reply prometheus/operator/base/servicemonitor.yaml]: Ok, fair enough

**Comment by vpavlin** [comment]
We merged the Spark Operator PR, so I rebased here in the web ui to be able to test it easily - seems to be working fine.

I am still not sure about the name `web`  for the port, but we'll see what other services have when we do a ODH wide check for monitoring

File: prometheus/operator/base/prometheus.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: prometheus
+  labels:
+    prometheus: k8s
+  namespace: prometheus
+spec:
+  replicas: 2
```
Comment (vpavlin) [inline prometheus/operator/base/prometheus.yaml:9]: Do we need 2 replicas by default?

Comment (nakfour) [reply prometheus/operator/base/prometheus.yaml:9]: No we don't

File: kfdef/kfctl_openshift.yaml [hunk 3]
```diff
@@ -51,7 +45,7 @@ spec:
       repoRef:
         name: manifests
         path: kafka/kafka
-    name: kafka-cluster
+    name: kafka-cluster
```
Comment (crobby) [inline kfdef/kfctl_openshift.yaml:48]: looks like maybe an unnecessary extra space showed up here

Comment (nakfour) [reply kfdef/kfctl_openshift.yaml]: deleted it

**Comment by vpavlin** [comment]
I am going to take a look at the READMEs for the components and try to come up with consistent templates, but since we do not have it right now, this one is fine.

Looks good, merging

**Pull request merged**